core/utils/bits.h

```diff
@@ -163,6 +163,24 @@ static inline void MaskBytes(uint8_t *buf, uint8_t const *mask,
   }
 }
 
+// TODO(melvinw): it would be nice if MaskTopN() supported blobs.
+// Returns a mask for the first `n` bits of an integral type T, i.e. starting
+// from the most significant bit toward the least significant.
+// For example: MaskTopN<uint32_t>(9) will return 0xF8000000
+template <typename T, typename = std::enable_if<std::is_integral<T>::value>>
+static inline T MaskTopN(size_t n) {
```

Comment:
The name Mask here might be ambiguous here; since its definition could be different depending on the contexts (masking on? masking off?). How about more explicit names, such as `SetBitsHi()/SetBitsLo()` or so?

Reply:
Ah, I like `SetBits`-- will change